### PR TITLE
Remove index version and transport version from main endpoint response

### PR DESCRIPTION
**Pull request by rjernst**
The index and transport versions are low level details of how a node behaves in a cluster. They were recently added to the main endpoint response, but they are too low level and should be moved to another endpoint TBD.

This commit removes those versions from the main endpoint response. Due to the fact lucene version is now derived from index version, this commit also adds an explicit lucene version member to the main response. Note that while this effectively reverts #97386 and #96900, the reading/writing transport code remains temporarily.

### CODE REVIEW

File: modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java [hunk 1]
```diff
@@ -41,26 +40,30 @@ public class MainResponse extends ActionResponse implements ToXContentObject {
         super(in);
         nodeName = in.readString();
         version = Version.readVersion(in);
-        indexVersion = in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_031) ? IndexVersion.readVersion(in) : null;
-        transportVersion = in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_019) ? TransportVersion.readVersion(in) : null;
+
+        if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_034)) {
+            luceneVersion = in.readString();
+        } else {
+            if (in.getTransportVersion().onOrAfter(TransportVersion.V_8_500_031)) {
```
Comment (thecoop) [inline modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java:47]: Worth a few comments here explaining what's going on (and why we're reading values then throwing them away)

Comment (rjernst) [reply modules/rest-root/src/main/java/org/elasticsearch/rest/root/MainResponse.java:52]: Added in https://github.com/elastic/elasticsearch/pull/97675/commits/5ed9d69842d02099f26241b0d7215343151b8a5a

**Comment by rjernst** [comment]
@thecoop I'm not sure how we can modify the verify version constants tests going forward. The test is built on the assumption we keep a full mapping of version to lucene version for every version of ES, but that is no longer the case. You changed the test to infer the lucene version from index version, but by removing index version from the root endpoint, we no longer have that. I'm wondering if this test still makes sense to keep. WDYT?

**Comment by thecoop** [comment]
I think if the test doesn't make sense anymore it can go. We can still test that the lucene version it reports is the most recent one, but any more than that doesn't make sense.

**Comment by rjernst** [comment]
The lucene version it reports won't be the latest. This test is a bwc test that checks the version of lucene reported by older versions of ES matches what lucene version we have in our local code for that older version. I guess we can migrate this test to use a new diagnostic api that returns index version, but until then I don't think it makes sense since we have decoupled Version from Lucene version.

**Comment by rjernst** [comment]
I will disable the test here. I opened https://github.com/elastic/elasticsearch/issues/97736 so we don't forget to re-enable the test once we have a better way to find index version through an api.

**Review by thecoop — APPROVED** [review]

**Pull request merged**